sdk/containerregistry/azure-containerregistry/azure/containerregistry/_anonymous_exchange_client.py

```diff
@@ -3,15 +3,22 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-from typing import Any, Union, Optional
-
+from typing import Any, Optional, Union
+from azure.core.credentials import TokenCredential, AccessToken
 from ._exchange_client import ExchangeClientAuthenticationPolicy
 from ._generated import ContainerRegistry
-from ._generated.models._container_registry_enums import TokenGrantType
+from ._generated.models import TokenGrantType
 from ._helpers import _parse_challenge
 from ._user_agent import USER_AGENT
 
 
+class AnonymousAccessCredential(TokenCredential):
+    def get_token(
+        self, *scopes: str, claims: Optional[str] = None, tenant_id: Optional[str] = None, **kwargs
+    ) -> AccessToken:
```

Comment:
I'm tempted to say we should type this as returning `None` instead of `AccessToken`, since we won't actually return a token. That would violate the protocol and require a typing suppression though... I think that's fine, and that it would be better to suppress a warning for an edge case than to give an incorrect signature. @xiangyan99, do you have an opinion?

Reply:
What's the scenario of using AnonymousAccessCredential and what's the expected behavior?

Reply:
The anonymous client that uses the credential doesn't actually authenticate any requests, so the credential is intentionally unable to authenticate. The idea of using an actual TokenCredential here instead of using something like an empty string as the credential (which is now a required parameter to client construction) is to raise a more useful error in the off chance that token requests are made

Reply:
Here the one in .Net: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/ContainerRegistryAnonymousAccessCredential.cs#L14 I'm creating a similar one in Python to fix the mypy error about credential parameter type mismatching.
The return type should be same or a sub class of parent return type - `AccessToken`, using `None` will have override error in mypy.

Reply:
Got it -- it shouldn't be used anyway, so aligning with what .NET already has makes sense to me 🙂 Thanks for looking into that!